Sources/TuistCore/Utils/BinaryLocator.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import TSCBasic
+import TuistSupport
+
+enum BinaryLocatorError: FatalError, Equatable {
+    case swiftDocNotFound
+
+    var description: String {
+        switch self {
+        case .swiftDocNotFound:
+            return "Couldn't find the swift-doc binary."
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .swiftDocNotFound:
+            return .bug
+        }
+    }
+}
+
+/// Protocol that defines the interface to locate the tuist binary in the environment.
+public protocol BinaryLocating {
+    /// Returns the path to the swift-doc binary.
+    func swiftDocPath() throws -> AbsolutePath
+}
+
+public final class BinaryLocator: BinaryLocating {
+    public init() {}
+
+    public func swiftDocPath() throws -> AbsolutePath {
+        #if DEBUG
+            // Used only for debug purposes to find templates in your tuist working directory
+            // `bundlePath` points to tuist/Templates
```

Comment:
Apologies if this was raised / discussed before, I may have missed it - Looks like swift doc offers a library

https://github.com/SwiftDocOrg/swift-doc/blob/master/Package.swift#L10

Integrating with the library (like the approach taken with `SwiftGen`) could alleviate the need to manage additional binaries and may be easier to interop with - were there issues with that approach? (e.g. conflicts is transitive dependencies?)

Reply:
Agree @fortmarek !! We worked on this with @pepibumur at the beginning to quickly draft v0, but this looks like something that will break quite easily.

I like that approach, but I wonder if moving `BinaryLocator` to `TuistKit` would comply with the architecture. I am not sure if it should be in `TuistCore` either 😅 . My bet would be `TuistSupport`. 

In my ex-job, we tackled this kind of issues generating some code. If we group all the binaries under the same folder, we can generate a BinaryLocator that will expose one method per binary.

What do you all think?

Reply:
@kwridan using the package was our first approach, but we saw that there are some types under [swift-doc/Supporting types](https://github.com/SwiftDocOrg/swift-doc/tree/master/Sources/swift-doc/Supporting%20Types) so we were not 100% that it was going to work. 

I am totally up for using the package on v2.

Reply:
I think `TuistCore` would be a good option as it is a place where we should place things that are tuist-specific, in contrast with `TuistSupport` which should be tuist-agnostic. 

I'd love to see the package itself being used, but it can totally wait for the time being 🙂

Reply:
I've tested the approach with `AbsolutePath(TuistCommand.processArguments()!.first!)` but I've realised that we're copying swift-doc when running `rake release`, so it won't find it under `/Library/Developer/Xcode/DerivedData/tuist-baoxwmscomgyesaccklmkkzhrvea/Build/Products/Debug/`.

Reply:
I guess we can tackle the whole binary locator approach in another PR, what do you think?

Reply:
Yeah, if you plan to use `swift-doc` library anyway, let's not hold this PR back just because of this 🙂